internal/pkg/diagnostics/diagnostics.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package diagnostics
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"runtime/pprof"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/elastic/elastic-agent/internal/pkg/release"
+)
+
+// Hook is a hook that gets used when diagnostic information is requested from the Elastic Agent.
+type Hook struct {
+	Name        string
+	Filename    string
+	Description string
+	ContentType string
+	Hook        func(ctx context.Context) []byte
+}
+
+// Hooks is a set of diagnostic hooks.
+type Hooks []Hook
+
+// GlobalHooks returns the global hooks that can be used at anytime with no other references.
+func GlobalHooks() Hooks {
+	return Hooks{
+		{
+			Name:        "version",
+			Filename:    "version.txt",
+			Description: "version information",
+			ContentType: "application/yaml",
+			Hook: func(_ context.Context) []byte {
+				v := release.Info()
+				o, err := yaml.Marshal(v)
+				if err != nil {
+					return []byte(fmt.Sprintf("error: %q", err))
+				}
+				return o
+			},
+		},
+		{
+			Name:        "goroutine",
+			Filename:    "goroutine.txt",
+			Description: "stack traces of all current goroutines",
+			ContentType: "plain/text",
+			Hook:        pprofDiag("goroutine"),
+		},
+		{
+			Name:        "heap",
+			Filename:    "heap.txt",
+			Description: "a sampling of memory allocations of live objects",
+			ContentType: "plain/text",
+			Hook:        pprofDiag("heap"),
+		},
+		{
+			Name:        "allocs",
+			Filename:    "allocs.txt",
+			Description: "a sampling of all past memory allocations",
+			ContentType: "plain/text",
+			Hook:        pprofDiag("allocs"),
+		},
+		{
+			Name:        "threadcreate",
+			Filename:    "threadcreate.txt",
+			Description: "stack traces that led to the creation of new OS threads",
+			ContentType: "plain/text",
+			Hook:        pprofDiag("threadcreate"),
+		},
+		{
+			Name:        "block",
+			Filename:    "block.txt",
+			Description: "stack traces that led to blocking on synchronization primitives",
+			ContentType: "plain/text",
+			Hook:        pprofDiag("block"),
+		},
+		{
+			Name:        "mutex",
+			Filename:    "mutex.txt",
+			Description: "stack traces of holders of contended mutexes",
+			ContentType: "plain/text",
+			Hook:        pprofDiag("mutex"),
+		},
+	}
+}
+
+func pprofDiag(name string) func(context.Context) []byte {
+	return func(_ context.Context) []byte {
```

Comment:
Is it worth sharing the same implementation as the client in case we find something wrong with it in the future? 

https://github.com/elastic/elastic-agent-client/blob/c29f5dc424c21a2b5a6a6ba48b8fe8d20e2eb8cd/pkg/client/client_v2.go#L667-L677

Reply:
Its rather simple implementation, I think importing that code into this area doesn't seem necessary or any reason they should share code for something that is only 5 lines of code.